src/wmtk/SimplicialComplex.cpp

```diff
@@ -287,15 +291,62 @@ SimplicialComplex SimplicialComplex::link(const Mesh& m, const Simplex& s)
 
 SimplicialComplex SimplicialComplex::open_star(const Mesh& m, const Simplex& s)
 {
-    SimplicialComplex sc_clst = closed_star(m, s);
     SimplicialComplex sc(m);
-    sc.add_simplex(s);
-    for (const Simplex& ss : sc_clst.get_simplices()) {
-        if (ss.primitive_type() <= s.primitive_type()) {
-            continue;
-        }
-        if (simplices_w_boundary_intersect(m, s, ss)) {
-            sc.add_simplex(ss);
+    
+    constexpr PrimitiveType PV = PrimitiveType::Vertex;
+    constexpr PrimitiveType PE = PrimitiveType::Edge;
+    constexpr PrimitiveType PF = PrimitiveType::Face;
+    constexpr PrimitiveType PT = PrimitiveType::Tetrahedron;
+
+    auto sw = [&m](const Tuple& _t, const PrimitiveType& _ptype) {
+        return m.switch_tuple(_t, _ptype);
+    };
+
+    const long cell_dim = dynamic_cast<const TriMesh*>(&m) ? 2 : 3;
+    const PrimitiveType s_ptype = s.primitive_type();
```

Comment:
the previous dynamic cast to determine `cell_dim` and this `s_ptype` are redundant right?